modules/500-openvpn/images/openvpn/entrypoint/entrypoint.go

```diff
@@ -0,0 +1,293 @@
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = insertUnique(iptablesMgr, mangleOutputRestoreMark.Table, mangleOutputRestoreMark.Chain, mangleOutputRestoreMark.Rule, mangleOutputRestoreMark.Pos)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	rule := netlink.NewRule()
+	rule.Table = routeTable
+	rule.Priority = 1
+	rule.Mark = routeTable
+	err = netlink.RuleAdd(rule)
+
+	if err != nil {
+		if !os.IsExist(err) {
+			log.Fatal(err)
+		}
+	}
+
+	err = mknodDevNetTun()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = netLinkCreateTuntap(fmt.Sprintf("tun-%s", protocol), 1400)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	routeAdd(network, fmt.Sprintf("tun-%s", protocol), routeTable)
+
+	requiredFiles := []string{
+		"/etc/openvpn/certs/pki/ca.crt",
+		"/etc/openvpn/certs/pki/private/server.key",
+		"/etc/openvpn/certs/pki/issued/server.crt",
+		"/etc/openvpn/certs/pki/ta.key",
+		"/etc/openvpn/certs/pki/dh.pem",
+		"/etc/openvpn/certs/pki/crl.pem",
+	}
+	for _, path := range requiredFiles {
+		waitingForFile(path)
+	}
+
+	var args []string
+	args = append(args, "--config")
+	args = append(args, "/etc/openvpn/openvpn.conf")
+	args = append(args, "--proto")
+	args = append(args, protocol)
+	args = append(args, "--management")
+	args = append(args, "127.0.0.1")
+	args = append(args, mgmtport)
+	args = append(args, "--dev")
+	args = append(args, fmt.Sprintf("tun-%s", protocol))
+	log.Println(args)
+
+	cmd := exec.Command("/usr/sbin/openvpn", args...)
+	cmdReader, err := cmd.StdoutPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+	scanner := bufio.NewScanner(cmdReader)
+	go func() {
+		for scanner.Scan() {
+			log.Println(scanner.Text())
+		}
+	}()
+	if err := cmd.Start(); err != nil {
+		log.Fatal(err)
+	}
+	if err := cmd.Wait(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func insertUnique(iptablesMgr *iptables.IPTables, table, chain string, rule []string, pos int) error {
+	ok, err := iptablesMgr.Exists(table, chain, rule...)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		err := iptablesMgr.Insert(table, chain, pos, rule...)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func mknodDevNetTun() error {
+	_, err := os.Stat("/dev")
+	if os.IsNotExist(err) {
+		err := unix.Mount("dev", "/dev", "devtmpfs", unix.MS_NOSUID|unix.MS_NOEXEC|unix.MS_RELATIME, "size=10m,nr_inodes=248418,mode=755")
+		if err != nil {
+			return fmt.Errorf("error mounting %s to %s: %v", "dev", "/dev", err)
+		}
+	}
+
+	err = unix.Mkdir("/dev/net", 0753)
+	if err != nil {
+		if !os.IsExist(err) {
+			return fmt.Errorf("error create dir /dev/net: %v", err)
+		}
+	}
+
+	command := fmt.Sprintf("/bin/mknod")
+	var args []string
+	args = append(args, "/dev/net/tun")
+	args = append(args, "c")
+	args = append(args, "10")
+	args = append(args, "200")
+	cmd := exec.Command(command, args...)
+	_, err = cmd.CombinedOutput()
```

Comment:
we do not need output here. Please use `cmd.Run` instead `CombinedOutput`

Reply:
done